### PR TITLE
cpu/sam0_common: implement periph/dac

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

The sam0 MCUs all have a DAC peripheral.
The DAC has a resulution of 10 or 12 bits and can have one or two output channels.

The output pins are always hard-wired to PA2 for DAC0 and PA5 for DAC1 if it exists.

On the `same54-xpro` I would only get a max value of ~1V when using the internal reference, so I configured it to use an external voltage reference.

The external reference pin is hard-wired to PA3, so there you'll have to connect that to 3.3V to get results.

### Testing procedure

Run `tests/periph_dac` and connect a multimeter to PA02. (and PA05 if supported)
I had to increase `DELAY` to 10000 (10ms / step) for my meter to keep up.
If you have an oscilloscope you should see a square wave directly.

Only tested on `same54-xpro` and `saml10-xpro`.

### CODE REVIEW

File: cpu/sam0_common/periph/dac.c [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+#include "periph/dac.h"
+#include "periph/gpio.h"
+
+#include "byteorder.h"
+
+#ifndef DAC_CLOCK
+#define DAC_CLOCK   DAC_CLOCK_DEFAULT
+#endif
+
+#define DAC_VAL(in) (in >> (16 - DAC_RES_BITS))
```
Comment (dylad) [inline cpu/sam0_common/periph/dac.c:31]: I don't get this.
Why playing with the bit position here ?
In the current state, DAC_DACCTRL_LEFTADJ is set to 0 and we don't use dithering so we should write data to bits [0..9] or  [0..11]. there is no need for this macro, just assign the value to the register directly.

Comment (benpicco) [reply cpu/sam0_common/periph/dac.c:31]: The problem is that the DAC API assumes a full 16-bit input value.
So setting a value > 1023 will make the DAC overflow back to zero - you can observe this with `tests/periph_dac`.

On other DACs (e.g. lpc2387) you write the full 16-bit value but this being a 10 bit DAC, the lower (least significant) 6 bytes are ignored.
Here the most significant bytes would be ignored, which leads to wrong output values.

**Comment by benpicco** [comment]
> * SAML10/SAML11 DAC are connected to 32kHz GCLK ? Maybe it's time to allocate an unused GCLK to provide a faster clock.

Well it should be enough for the audio player (#13908 - didn't test that yet though), I think we shouldn't configure an extra clock just in case.

**Comment by dylad** [comment]
Aren't we suppose to configure PA03 to MUX_B using `gpio_init_mux` if we want to use it as external reference ?

**Comment by benpicco** [comment]
Yea you are right - it just worked™ without setting the mux, but it's probably better to be explicit.

File: cpu/sam0_common/periph/dac.c [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+ * @author      Benjamin Valentin <benpicco@beuth-hochschule.de>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "periph/dac.h"
+#include "periph/gpio.h"
+
+#include "byteorder.h"
```
Comment (dylad) [inline cpu/sam0_common/periph/dac.c:25]: Still needed ?

Comment (benpicco) [reply cpu/sam0_common/periph/dac.c]: Nope

**Comment by benpicco** [comment]
Thank you for testing this on the whole sam0 family! I currently only have a limited set of boards to test on.

**Comment by dylad** [comment]
Here we go !

**Pull request merged**